cli/src/pcluster/models/cluster.py

```diff
@@ -663,28 +657,47 @@ def start(self):
+        compute_fleet_status_manager = ComputeFleetStatusManager(self.name)
+        compute_fleet_status_manager.update_status(
+            ComputeFleetStatus.START_REQUESTED, ComputeFleetStatus.STARTING, ComputeFleetStatus.RUNNING
+        )
+
+    def enable_awsbatch_compute_environment(self):
+        """Enable AWS Batch compute environment."""
+        stack_status = self.stack.status
+        if "IN_PROGRESS" in stack_status:
+            raise BadRequestClusterActionError(
+                f"Cannot enable compute environment while stack is in {stack_status} status."
+            )
+        LOGGER.info("Enabling AWS Batch compute environment : %s", self.name)
+        try:
+            compute_resource = self.config.scheduling.queues[0].compute_resources[0]
+
+            AWSApi.instance().batch.enable_compute_environment(
+                ce_name=self.stack.batch_compute_environment,
+                min_vcpus=compute_resource.min_vcpus,
+                max_vcpus=compute_resource.max_vcpus,
+                desired_vcpus=compute_resource.desired_vcpus,
+            )
+        except Exception as e:
+            raise _cluster_error_mapper(e, f"Unable to enable Batch compute environment. {str(e)}")
+
     def stop(self):
         """Stop compute fleet of the cluster."""
         try:
             scheduler = self.config.scheduling.scheduler
+            stack_status = self.stack.status
+            if "FAILED" in stack_status:
+                LOGGER.warning("Cluster stack is in %s status. This operation might fail.", stack_status)
             if scheduler == "awsbatch":
-                LOGGER.info("Disabling AWS Batch compute environment : %s", self.name)
-                try:
-                    AWSApi.instance().batch.disable_compute_environment(ce_name=self.stack.batch_compute_environment)
-                except Exception as e:
-                    raise _cluster_error_mapper(e, f"Unable to disable Batch compute environment. {str(e)}")
-
+                self.disable_awsbatch_compute_environment()
             else:  # scheduler == "slurm"
-                stack_status = self.stack.status
-                if "IN_PROGRESS" in stack_status:
-                    raise ClusterActionError(f"Cannot stop compute fleet while stack is in {stack_status} status.")
-                if "FAILED" in stack_status:
-                    LOGGER.warning("Cluster stack is in %s status. This operation might fail.", stack_status)
-
-                compute_fleet_status_manager = ComputeFleetStatusManager(self.name)
-                compute_fleet_status_manager.update_status(
-                    ComputeFleetStatus.STOP_REQUESTED, ComputeFleetStatus.STOPPING, ComputeFleetStatus.STOPPED
-                )
+                self.stop_slurm_compute_fleet()
         except ComputeFleetStatusManager.ConditionalStatusUpdateFailed:
             raise ClusterActionError(
```

Comment:
BadRequestClusterActionError?
and same question for 
```
except ClusterActionError as e:
            raise e
```

Reply:
Done